unitex/src/fr/umlv/unitex/tfst/tagging/TaggingModel.java

```diff
@@ -105,27 +150,677 @@ protected void updateModel() {
 			}
 		}
 	}
-
-	/**
+	
+	/** @Yass
 	 * This method must be called when a new sentence automaton has been loaded.
+	 * This method resets the various arrays used in this class by applying all the graphBoxes, including new ones, in them.
 	 */
 	public void resetModel() {
 		final int n = zone.graphBoxes.size();
 		boxes = new TfstGraphBox[n];
 		taggingStates = new TaggingState[n];
 		factorization = new boolean[n];
 		renumber = new int[n];
+		//init 
+		tokens = new HashMap<Integer,ArrayList<String>>();
 		for (int i = 0; i < n; i++) {
 			boxes[i] = (TfstGraphBox) zone.graphBoxes.get(i);
-			if (boxes[i].type == GenericGraphBox.INITIAL)
+			if (boxes[i].type == GenericGraphBox.INITIAL) {
 				initialState = i;
+			}
 			else if (boxes[i].type == GenericGraphBox.FINAL)
 				finalState = i;
 			taggingStates[i] = TaggingState.NEUTRAL;
+			boxes[i].state = TaggingState.NEUTRAL;
+		}
+		updateNodes();
+		generateAlphabet();
+		generateTokensList();
+		zone.unSelectAllBoxes();
+		lstTok.clear();
+		lstPath.clear();
+		
+	}
+	
+	void generateAlphabet() {
+		if( regex == null || !alphabetFile.equals(ConfigManager.getManager().getAlphabet(null))) { 
+			// switching languages must also switch this
+			alphabetFile = ConfigManager.getManager().getAlphabet(null);
+			try {
+				StringBuilder alphabet= new StringBuilder();
+				
+				BufferedReader br = new BufferedReader( new InputStreamReader( new FileInputStream(alphabetFile) , "UTF16"));
+				String line; 
+				while ((line = br.readLine()) != null && !line.equals("\r\n")) {
+					if( line.startsWith("#")) {
+						alphabet.append(line.charAt(0)).append("-").append(line.charAt(1));
+					}		
+					alphabet.append(line.trim());
+				}
+				StringBuilder regexBuilder = new StringBuilder();
+				regexBuilder.append("(<E>)|([").append(alphabet.toString())
+					.append("]+|[^").append(alphabet.toString()).append("]+)");
+				regex = regexBuilder.toString();
+			} catch (IOException e) {
+				 //TODO Auto-generated catch block
+				System.out.println("exception");
+			} finally {}	
+		}	
+	}
+	
+	/*
+	 * for each box we will start at it's starting bound, split the inside based on delimiters and add them to the hashmap. 
+	 * if 2 words split by a comma are found, they will be stored subsequently in the hashmap at the starting bound's value, then +1 and so on. 
```

Comment:
The description in this line is wrong and is about a detail which is not so relevant